packages/components/src/internal/components/user/APIKeysPanel.tsx

```diff
@@ -234,12 +233,9 @@ const APIKeysPanelBody: FC<APIKeysPanelBodyProps & InjectedQueryModels> = props
         }
     }, []);
 
-    const onApiKeyCreate = useCallback(
-        () => {
-            actions?.loadModel(model?.id, true, true);
-        },
-        [actions, model?.id]
-    );
+    const onApiKeyCreate = useCallback(() => {
+        actions.onModelChange(model?.id, { changeType: ChangeType.add });
```

Comment:
This should probably bail if `model` is not defined as passing in undefined to `actions.onModelChange` will result in errors. I realize this likely will never happen due to impersonated users not being able to generate API keys.

<img width="569" alt="image" src="https://github.com/user-attachments/assets/6afc794d-ac7f-4300-a93e-73730f6b26e3" />

Reply:
I don't think that is a good idea, I would rather we refactor this component to always have a model present so that class of error does not exist. We should be handling the impersonation case before the model is created.

Reply:
I have this refactor done locally, however it's a pretty big change, so I am going to open a separate PR for it.